actstream/migrations/0001_initial.py

```diff
@@ -6,12 +6,7 @@
 import django.utils.timezone
 from django.conf import settings
 
-from actstream.settings import USE_JSONFIELD
-
-if USE_JSONFIELD:
-    from jsonfield_compat.fields import JSONField as DataField
-else:
-    DataField = models.TextField
+from actstream.jsonfield import DataField
```

Comment:
can we check this is backward compatible? any migration changes needed?

Reply:
This is backwards compatible, as the default behaviour is unchanged and the model itself untouched, nothing new has been added or modified. I basically just moved the import and check of `Use_JSONFIELD` from this migration file to the `actstream/jsonfield.py`. That module is responsible for a) deciding if we should keep a `data` field at all and b) decide on an implementation of that field type.

The only change in behaviour occurs when running initial migrations for the first time with `USE_JSONFIELD = True` and `django-mysql` installed, while `django-jsonfield-compat` is **not installed**. In that case the `Action` model will get to keep a `data` field that is of type `django_mysql.models.json.JSONField` instead of `jsonfield.fields.JSONField`.

The different JSONFields are compatible to each other, as the only interaction that `actstream` is doing is assigning a dict to that field (`newaction.data = kwargs`), so I don't think we have to worry about any API incompatibilities.